pkg/advisory/discover_aliases.go

```diff
@@ -0,0 +1,150 @@
+package advisory
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"sort"
+
+	"github.com/samber/lo"
+	"github.com/wolfi-dev/wolfictl/pkg/configs"
+	v2 "github.com/wolfi-dev/wolfictl/pkg/configs/advisory/v2"
+	"github.com/wolfi-dev/wolfictl/pkg/vuln"
+)
+
+// DiscoverAliasesOptions is the set of options for the DiscoverAliases
+// function.
+type DiscoverAliasesOptions struct {
+	// AdvisoryDocs is the Index of advisory documents on which to operate.
+	AdvisoryDocs *configs.Index[v2.Document]
+
+	// AliasFinder is the alias finder to use for discovering aliases for the given
+	// vulnerabilities.
+	AliasFinder AliasFinder
+
+	// SelectedPackages is the list of packages to operate on. If empty, all
+	// packages will be operated on.
+	SelectedPackages []string
+}
+
+// DiscoverAliases queries external data sources for aliases for the
+// vulnerabilities described in the selected advisories and updates the advisory
+// documents with the discovered aliases.
+func DiscoverAliases(ctx context.Context, opts DiscoverAliasesOptions) error {
+	documents := opts.AdvisoryDocs.Select().Configurations()
+
+	// If the caller selected specific packages (by name), filter the set of
+	// advisory documents to just those packages.
+	if len(opts.SelectedPackages) > 0 {
+		documents = lo.Filter(documents, func(doc v2.Document, _ int) bool {
+			return slices.Contains(opts.SelectedPackages, doc.Package.Name)
+		})
+	}
+
+	for _, doc := range documents {
+		for _, adv := range doc.Advisories {
+			// If the advisory ID is already a CVE ID, do a lookup and ensure the discovered
+			// aliases are present in the advisory's list of aliases.
+			if vuln.RegexCVE.MatchString(adv.ID) {
```

Comment:
I wonder if we could extract the logic for each of the `if` statements in functions. It gets a bit hard to read!

This pattern also reminds me of the switch with no condition, e.g. this might read better:

```
switch {
case vuln.RegexCVE.MatchString(adv.ID):
    // logic
case vuln.RegexGHSA.MatchString(adv.ID):
    // logic
default:
    // whatever
}
```

Reply:
Great suggestion! Updated!